release/RELEASE.md

```diff
@@ -1,149 +1,157 @@
 # Making a Release
 
-Valid commits for a release should come from either the `main` branch or one
-of the support `release/a.b.x` branches (e.g. `release/1.0.x` branch is for
-patches we backport to the 1.0 release branch).
+Please read this document carefully.
+
+At a high level, a release goes through the following steps. All releases must
+use the same version number for stable releases; there is more leeway for
+snapshots and RCs.
+
+1. A "daml repo" release, orchestrated from this repository.
+2. A "canton repo" release, where the code in the release commit must have a
+   dependency on the artifacts produced by step 1.
+3. An "assembly repo" release, which combines the daml and canton release
+   artifacts and creates the GitHub Release on the daml repo.
+4. A "daml-finance repo" release. This is independant of the other steps on a
+   technical level.
+5. A "docs repo" release to publish documentation on
+   [docs.daml.com](https://docs.daml.com).
+6. Manual tests and approval of the final release artifacts.
+
+Note that these are the technical steps to create the release artifacts. The
+release process as a whole is quite a bit larger and involves synchronization
+between various teams about exactly what we want to include in a release, when
+we want to release it, communication to various stakeholders, etc. For details
+on the broader process, see the [release planning] document.
+
+For the more detailed technical steps, we need to distinguish between three
+cases:
+
+a. A new minor release.
+b. A patch release on an existing minor version.
+c. The weekly snapshot.
+
+The way in which we get to the final artifact differs based on the three cases
+above, but the manual testing steps are always the same. As such, they have
+their own section at the end of this document.
+
+## Minor Release
+
+For a minor release, we will usually set a target date and a target scope. When
+we get "close enough" to the target scope or date, we create the release
+branches for the future release, branching off from `main`. There is no hard rule about
+when this step happens, and there may be a few days between repos.
+
+In the [daml] repo, the release branch for minor version `A.B` must be named
+`release/A.B.x`, e.g.  `release/2.7.x`. In the [canton] repo, the release
+branch is names `release-line-A.B`, e.g. `release-line-2.7`. In both repos, we
+have special "branch protection rules" set up in GitHub for branches with those
+names.
+
+> Note: This also means you should not create branches with those (or similar)
+> names if they are not meant to be release branches.
+
+When the release branches contain all the required changes and there are no
+blockers left, we create a release candidate (RC). This step can be repeated
```

Comment:
Note: in Canton we tend to do the opposite (first the rc1 and then the creation of the release-line branch). But I don't think that this is worth mentioning in this guide

Reply:
Historically the reason we don't do that on the daml repo is that you then need to have a period where you tell people not to merge stuff if they don't want it in the release.

Reply:
Yes, that's probably the best strategy to first create the release line branch and then the rc. We used to be limited by the tooling but we improved it, let's see for 2.7